src/sa_basis.jl

```diff
@@ -1,160 +1,233 @@
-Base.convert(::Type{M}, b::SemisimpleSummand) where {M<:AbstractMatrix} =
-    convert(M, basis(b))
+function image_basis(α::AlgebraElement)
+    image, pivots = image_basis!(matrix_projection(α))
+    return image[1:length(pivots), :]
+end
 
-multiplicity(b::SemisimpleSummand) = b.multiplicity
+function image_basis(hom::InducedActionHomomorphism, α::AlgebraElement)
+    image, pivots = image_basis!(matrix_projection(hom, α))
+    return image[1:length(pivots), :]
+end
 
-function degree(b::SemisimpleSummand)
-    d, r = divrem(size(basis(b), 1), multiplicity(b))
-    @assert iszero(r)
-    return d
+struct DirectSummand{T, M<:AbstractMatrix{T}} <: AbstractMatrix{T}
+    basis::M
+    multiplicity::Int
+    degree::Int
+    simple::Bool
 end
 
-function extended_characters(::Type{T}, G::Group, basis, action) where {T}
-    chars = characters_dixon(T, G)
-    ehom = ExtensionHomomorphism(action, basis)
-    return ehom(chars) # result: Cyclotomic{T} valued Characters
+StarAlgebras.basis(ds::DirectSummand) = ds.basis
+issimple(ds::DirectSummand) = ds.simple
+degree(ds::DirectSummand) = ds.degree
+multiplicity(ds::DirectSummand) = ds.multiplicity
+
+Base.size(ds::DirectSummand) = size(ds.basis)
+Base.@propagate_inbounds Base.getindex(ds::DirectSummand, i...) =
+    basis(ds)[i...]
+
+function affordable_real(
+    irreducible_characters,
+    multiplicities=fill(1, length(irreducible_characters)),
+)
+    irr_real = similar(irreducible_characters, 0)
+    mls_real = similar(multiplicities, 0)
+    for (i, χ) in pairs(irreducible_characters)
+        ι = frobenius_schur(χ)
+        if abs(ι) == 1 # real or quaternionic
+            @debug "real/quaternionic:" χ
+            push!(irr_real, χ)
+            push!(mls_real, multiplicities[i])
+        else # complex one...
+            cχ = conj(χ)
+            k = findfirst(==(cχ), irreducible_characters)
+            @assert !isnothing(k)
+            @debug "complex" χ conj(χ)=irreducible_characters[k]
+            if k > i # ... we haven't already observed a conjugate of
+                @assert multiplicities[i] == multiplicities[k]
+                push!(irr_real, χ + cχ)
+                push!(mls_real, multiplicities[i])
+            end
+        end
+    end
+
+    return irr_real, mls_real
 end
 
 """
-    symmetry_adapted_basis([T::Type,] G::AbstractPermutationGroup[, S=Rational{Int}])
-Compute a basis for the linear space `ℝⁿ` which is invariant under the symmetry of `G`.
+    symmetry_adapted_basis([T::Type,] G::AbstractPermutationGroup[, S=Rational{Int};
+        semisimple=false])
+Compute a basis for the linear space `ℝⁿ` (where `n = degree(G)`) which is
+invariant under the symmetry of `G`.
 
-The permutation group is acting naturally on `1:degree(G)`. The coefficients of
-the invariant basis are returned in (orthogonal) blocks corresponding to irreducible
-characters of `G`.
+The coefficients of the invariant basis are returned in (orthogonal) blocks
```

Comment:
Aren't they orthogonal only if `T<:AbstractFloat` ?

Reply:
The blocks are always pairwise orthogonal. The vectors within each block are orthogonal only when `T<:AbstractFloat`

Reply:
Maybe that could be clearer in the docstring